### ISSUE TITLE
Some development dependencies are not versioned

### ISSUE THREAD

**Issue by AlekSi** [prometheus/prometheus#3022]
**What did you do?**
```
git checkout dev-2.0
scripts/genproto.sh
git status
```

**What did you expect to see?**
```
nothing to commit, working tree clean
```

**What did you see instead? Under which circumstances?**
```
Changes not staged for commit:
	modified:   documentation/dev/api/swagger.json
	modified:   prompb/rpc.pb.go
	modified:   prompb/rpc.pb.gw.go
	modified:   prompb/types.pb.go
```

This script uses https://github.com/gogo/protobuf and https://github.com/grpc-ecosystem/grpc-gateway from GOPATH, they are not vendored. This means that every contributor who touches that part (like me) will create a different version of generated files. The changes are not trivial.

I propose to vendor those dependencies and use them. What do you think?

**Comment by brian-brazil** [comment]
Those dependencies have disto and arch specific non-Go dependencies themselves, so I don't think it's possible to vendor them.

**Comment by AlekSi** [comment]
Well, `protoc` version is checked by `genproto.sh` script. What else?

**Comment by gouthamve** [comment]
We went through this issue in our bug scrub and this seems to be fixed in c3434a39b761cff8bb6e444fecaf312c5df2bd68

Please let us know if it isn't the case.
